### PR TITLE
Preparing for the summer: baseball and dodgeballs!

### PR DESCRIPTION
**Pull request by Henri215**
## What Does This PR Do

**Supply crate:** Jersey crate now has baseball instead of beach ball. Included 2 dodgeballs to laser tag crate.
**Description:** Added to beach ball.
**Size:** Beach ball (tiny --> normal), baseball (tiny --> small).

Dodgeball has been added to lasertag closets.
Dodgeball now works like laser tag rifles, it will inflict knockdown (6 SECONDS) if it hits someone wearing the armour.

Moved basketball, dodgeball, baseball, beach ball and hoop to the new "sports.dm".
## Why It's Good For The Game

The baseball wasnt being used anywhere in the game and it makes no sense to have the beach ball inside the Jersey crate.
There was no non event variation of the dodgeball. Its inclusion to the game, linked to the tag armour system, is going to incentivize people running sports events.
## Images of changes

https://user-images.githubusercontent.com/77684085/200198536-cd665dd1-7e73-4f88-b00b-0ac2ac225bb3.mp4

## Testing

- Spawned all balls and tested the dodgeball with tag armours.
- Tried to steal holodeck balls, failed.
## Changelog
:cl:
tweak: Jersey crate: now has baseball instead of beach ball.
tweak: Laser tag crate: 2 dodgeballs added to it.
tweak: Beach ball size is now NORMAL.
tweak: Dodgeball knockdowns if the target is wearing laser tag armour.
/:cl:

### CODE REVIEW

File: code/datums/supplypacks.dm [hunk 1]
```diff
@@ -1411,7 +1411,9 @@ GLOBAL_LIST_INIT(all_supply_groups, list(SUPPLY_EMERGENCY,SUPPLY_SECURITY,SUPPLY
 
 /datum/supply_packs/misc/lasertag
 	name = "Laser Tag Crate"
-	contains = list(/obj/item/gun/energy/laser/tag/red,
+	contains = list(/obj/item/beach_ball/dodgeball,
+					/obj/item/beach_ball/dodgeball,
+					/obj/item/gun/energy/laser/tag/red,
 					/obj/item/gun/energy/laser/tag/red,
```
Comment (Contrabang) [inline code/datums/supplypacks.dm:1417]: This is gonna conflict with the economy rework

File: code/game/objects/items/sport.dm [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	var/suit_types = list(/obj/item/clothing/suit/redtag, /obj/item/clothing/suit/bluetag)
+
+/obj/item/beach_ball/dodgeball/throw_impact(atom/hit_atom)
+	. = ..()
+	var/mob/living/carbon/human/M = hit_atom
+	if(ishuman(hit_atom) && istype(M.wear_suit) && (M.wear_suit.type in suit_types))
+		if((M.r_hand == src) || (M.l_hand == src))
+			return
+		playsound(src, 'sound/items/dodgeball.ogg', 50, 1)
+		M.adjustStaminaLoss(100)
```
Comment (Contrabang) [inline code/game/objects/items/sport.dm:39]: Instant stuns may not be the best idea... but balance decision.

File: code/game/objects/items/sport.dm [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	return
```
Comment (Contrabang) [inline code/game/objects/items/sport.dm:40]: This hanging return isnt needed, you can get rid of it.
```suggestion

```

File: code/game/objects/items/sport.dm [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	name = "beach ball"
+	icon = 'icons/misc/beach.dmi'
+	desc = "An inflatable ball of fun, enjoyed on many beaches."
+	icon_state = "ball"
+	item_state = "beachball"
+	density = FALSE
+	anchored = FALSE
+	w_class = WEIGHT_CLASS_NORMAL
+	force = 0.0
+	throwforce = 0.0
```
Comment (Contrabang) [inline code/game/objects/items/sport.dm:11]:
```suggestion
	force = 0
	throwforce = 0
```
More readable imo

File: code/game/objects/items/sport.dm [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+	throw_speed = 1
+	throw_range = 20
+	flags = CONDUCT
```
Comment (Contrabang) [inline code/game/objects/items/sport.dm:14]: Not asking to change anything here, but why are beach balls conductive lmao 😂

**Review by Contrabang — COMMENTED** [review]
Just a few review things

File: code/datums/supplypacks.dm (hunk 1 above)
Comment (Henri215) [reply code/datums/supplypacks.dm:1417]: I could add it to the crate later but...exchanging the beach ball for the baseball one is necessary, no problem on being on hold.

**Comment by PacifistDalek** [comment]
This is a fantastic idea!

**Comment by Sayrbee** [comment]
the dodge ball should make that pi-toink sound

**Comment by Henri215** [comment]
> the dodge ball should make that pi-toink sound

Maybe, but it already has a unique sound effect when someone is hit by it.

File: code/game/objects/items/sport.dm [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+	density = TRUE
+	pass_flags = LETPASSTHROW
+
+/obj/structure/holohoop/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/grab) && get_dist(src, user) <= 1)
+		var/obj/item/grab/G = W
+		if(G.state < GRAB_AGGRESSIVE)
+			to_chat(user, "<span class='warning'>You need a better grip to do that!</span>")
+			return
+		G.affecting.loc = src.loc
```
Comment (Sirryan2002) [inline code/game/objects/items/sport.dm:64]:
```suggestion
		G.affecting.forceMove(loc)
```

File: code/game/objects/items/sport.dm [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+	item_state = "baseball"
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/beach_ball/dodgeball
+	name = "dodgeball"
+	desc = "Used for playing the most violent and degrading of childhood games. This one is connected to the laser tag armour system."
+	icon = 'icons/obj/basketball.dmi'
+	icon_state = "dodgeball"
+	item_state = "dodgeball"
+	var/suit_types = list(/obj/item/clothing/suit/redtag, /obj/item/clothing/suit/bluetag)
```
Comment (Sirryan2002) [inline code/game/objects/items/sport.dm:30]:
```suggestion
	var/list/suit_types = list(/obj/item/clothing/suit/redtag, /obj/item/clothing/suit/bluetag)
```

File: code/game/objects/items/sport.dm [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+/obj/item/beach_ball/dodgeball/throw_impact(atom/hit_atom)
+	. = ..()
+	var/mob/living/carbon/human/M = hit_atom
+	if(ishuman(hit_atom) && istype(M.wear_suit) && (M.wear_suit.type in suit_types))
```
Comment (Sirryan2002) [inline code/game/objects/items/sport.dm:35]: did you mean to just check to see if `M.wear_suit` exists, it should never not be that type unless there's a bug elsewhere

File: code/game/objects/items/sport.dm [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+		if((M.r_hand == src) || (M.l_hand == src))
```
Comment (Sirryan2002) [inline code/game/objects/items/sport.dm:36]:
```suggestion
		if(M.r_hand == src || M.l_hand == src)
```

File: code/game/objects/items/sport.dm (hunk 8 above)
Comment (Henri215) [reply code/game/objects/items/sport.dm:35]: Thats what i meant, doing like this is better?
image

Comment (Sirryan2002) [reply code/game/objects/items/sport.dm:35]: Yes that should work

**Review by Sirryan2002 — APPROVED** [review]

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**